### PR TITLE
include broker configuration when we validate the kafka config

### PR DESCRIPTION
**Pull request by elindsey**
During purification we do a shallow validation of the kafka config to
suss out any major issues with authentication early on.
Previously we didn't include the bootstrap.servers parameter when we did
this validation, resulting in a logged warning like:

```
INFO librdkafka: CONFWARN [thrd:app]: No bootstrap.servers configured: client will not be able to connect to Kafka cluster
```

This warning was benign since the config is never used to build a kafka
client, but it is extremely misleading when trying to debug unrelated
kafka issues.

This change includes the bootstrap.servers config parameter to remove
the misleading warning.

Related to https://github.com/MaterializeInc/materialize/issues/4939

---
This change is [Reviewable](https://reviewable.io/reviews/materializeinc/materialize/5821)

### CODE REVIEW

**Review by benesch — APPROVED** [review]

**Comment by benesch** [comment]
I think we can optimistically close #4939, or are you not confident that this solves all the log spam?

**Comment by elindsey** [comment]
I'm confident it resolves the log spam! But I wanted to wait until merged to close 🙂

**Pull request merged**
